packages/astro/src/runtime/server/render/astro/render.ts

```diff
@@ -125,6 +160,54 @@ export async function renderToReadableStream(
 	});
 }
 
+export async function renderToAsyncIterable(
+	result: SSRResult,
+	componentFactory: AstroComponentFactory,
+	props: any,
+	children: any,
+	isPage = false,
+	route?: RouteData
+): Promise<AsyncIterable<Uint8Array>> {
```

Comment:
Question: Since we know strings as faster to deal with, could the async iterable return strings instead? Not sure if Node supports that.

Reply:
It might, but why do you think strings are faster to deal with? That's not obvious to me at least. In SSG everything gets turned into an array buffer (we call `response.arrayBuffer()`), and in dev + ssr prod everything gets turned into bytes too. So it makes sense to me to stay in bytes. Unless the string concatenation part is much faster.

Once the PR is ready I'd be happy to test a second implementation to see.

Reply:
I mentioned strings are faster in relation to my comment above, where the strings don't have to be encoded and decoded again. The default/base type of the chunks are strings. And I guess theoretically speaking concatenating strings will be faster than with bytes: https://github.com/withastro/astro/blob/6b4f77c100da3401f1749184e15b5218c947c70c/packages/astro/src/runtime/server/render/astro/render.ts#L80-L92

If we want to deal with bytes, we could concat the strings and then encode them as bytes in one step.